File: 0704-binary-search/0704-binary-search.go
import "math"

func search(nums []int, target int) int {
	left := 0
	right := len(nums) - 1
	mid := int(math.Round(float64((left + right) / 2)))

	for left <= right {
		if nums[mid] == target {
			return mid
		}

		if nums[mid] < target {
			left = mid + 1
			mid = int(math.Round(float64((left + right) / 2)))
			continue
		}

		if nums[mid] > target {
			right = mid - 1
			mid = int(math.Round(float64((left + right) / 2)))
			continue
		}
	}

	return -1
}

File: 0704-binary-search/0704-binary-search.ts
function search(nums: number[], target: number): number {
  let left = 0;
  let right = nums.length - 1;
  let mid = Math.round((left + right) / 2);

  while (left <= right) {
    if (nums[mid] === target) {
      return mid;
    }

    if (nums[mid] > target) {
      right = mid - 1;
      mid = Math.round((left + right) / 2);
      continue;
    }

    if (nums[mid] < target) {
      left = mid + 1;
      mid = Math.round((left + right) / 2);
      continue;
    }
  }

  return -1;
}